tests/framework/extensions/provisioning/creates.go

```diff
@@ -211,36 +215,61 @@ func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvide
 
 	checkFunc := clusters.IsProvisioningClusterReady
 	var command string
-	for key, node := range nodes {
-		logrus.Infof("Execute Registration Command for node %s", node.NodeID)
-		if strings.Contains(rolesPerNode[key], "windows") {
+	totalNodesObserved := 0
+	for poolIndex, poolRole := range rolesPerPool {
+		if strings.Contains(poolRole, "windows") {
+			totalNodesObserved += int(quantityPerPool[poolIndex])
 			continue
 		}
-		logrus.Infof("Linux pool detected, using bash")
-		command := fmt.Sprintf("%s %s --address %s", token.InsecureNodeCommand, rolesPerNode[key], node.PublicIPAddress)
-		logrus.Infof("Command: %s", command)
-		output, err := node.ExecuteCommand(command)
-		if err != nil {
-			return nil, err
+		for nodeIndex := 0; nodeIndex < int(quantityPerPool[poolIndex]); nodeIndex++ {
+			node := nodes[totalNodesObserved+nodeIndex]
+
+			logrus.Infof("Execute Registration Command for node %s", node.NodeID)
+			logrus.Infof("Linux pool detected, using bash...")
+
+			command = fmt.Sprintf("%s %s", token.InsecureNodeCommand, poolRole)
+			if clustersConfig.MachinePools[poolIndex].IsSecure {
+				command = fmt.Sprintf("%s %s", token.NodeCommand, poolRole)
+			}
+			command = createRegistrationCommand(command, node.PublicIPAddress, node.PrivateIPAddress, clustersConfig.MachinePools[poolIndex])
+			logrus.Infof("Command: %s", command)
+
+			output, err := node.ExecuteCommand(command)
+			if err != nil {
+				return nil, err
+			}
+			logrus.Infof(output)
 		}
-		logrus.Infof(output)
+		totalNodesObserved += int(quantityPerPool[poolIndex])
 	}
+
 	err = wait.WatchWait(result, checkFunc)
 	if err != nil {
 		return nil, err
 	}
+	totalNodesObserved = 0
+	for poolIndex := 0; poolIndex < len(rolesPerPool); poolIndex++ {
```

Comment:
kinda of a nit but why have the for loop like this rather than 
`for poolIndex, poolRole := range rolesPerPool `

Reply:
mostly because I don't plan on needing poolRole here and it looked better when I was writing this to match the nested loop's format but if you want it with range I can use that here

Reply:
no biggie, just curious. Let's leave it.